src/PowerShellEditorServices/Services/DebugAdapter/BreakpointService.cs

```diff
@@ -0,0 +1,293 @@
+                    // move onto the next breakpoint.
+                    if (actionScriptBlock == null)
+                    {
+                        configuredBreakpoints.Add(breakpoint);
+                        continue;
+                    }
+
+                    psCommand.AddParameter("Action", actionScriptBlock);
+                }
+            }
+
+            // If no PSCommand was created then there are no breakpoints to set.
+            if (psCommand != null)
+            {
+                IEnumerable<Breakpoint> setBreakpoints =
+                    await _powerShellContextService.ExecuteCommandAsync<Breakpoint>(psCommand);
+                configuredBreakpoints.AddRange(
+                    setBreakpoints.Select(CommandBreakpointDetails.Create));
+            }
+
+            return configuredBreakpoints;
+        }
+
+        /// <summary>
+        /// Clears all breakpoints in the current session.
+        /// </summary>
+        public async Task RemoveAllBreakpointsAsync(string scriptPath = null)
+        {
+            try
+            {
+                if (VersionUtils.IsPS7OrGreater)
+                {
+                    foreach (Breakpoint breakpoint in BreakpointApiUtils.GetBreakpoints(
+                            _powerShellContextService.CurrentRunspace.Runspace.Debugger,
+                            _debugStateService.RunspaceId))
+                    {
+                        if (scriptPath == null || scriptPath == breakpoint.Script)
+                        {
+                            BreakpointApiUtils.RemoveBreakpoint(
+                                _powerShellContextService.CurrentRunspace.Runspace.Debugger,
+                                breakpoint,
+                                _debugStateService.RunspaceId);
+                        }
+                    }
+
+                    return;
+                }
+
+                // Legacy behavior
+
+                PSCommand psCommand = new PSCommand();
+                psCommand.AddCommand(@"Microsoft.PowerShell.Utility\Get-PSBreakpoint");
+                psCommand.AddCommand(@"Microsoft.PowerShell.Utility\Remove-PSBreakpoint");
+
+                await _powerShellContextService.ExecuteCommandAsync<object>(psCommand);
+            }
+            catch (Exception e)
+            {
+                _logger.LogException("Caught exception while clearing breakpoints from session", e);
+            }
+        }
+
+        public async Task RemoveBreakpointsAsync(IEnumerable<Breakpoint> breakpoints)
+        {
+            if (VersionUtils.IsPS7OrGreater)
+            {
+                foreach (Breakpoint breakpoint in breakpoints)
+                {
+                    BreakpointApiUtils.RemoveBreakpoint(
+                        _powerShellContextService.CurrentRunspace.Runspace.Debugger,
+                        breakpoint,
+                        _debugStateService.RunspaceId);
+
+                    switch (breakpoint)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4598462)